rq/worker.py

```diff
@@ -783,6 +783,7 @@ def handle_job_success(self, job, queue, started_job_registry):
                     # if dependencies are inserted after enqueue_dependents
                     # a WatchError is thrown by execute()
                     pipeline.watch(job.dependents_key)
+                    job.set_status(JobStatus.FINISHED, pipeline=pipeline)
```

Comment:
Is there a reason why `job.set_status` is moved here? If the job is deleted right away, this operation doesn't need to be called.

Reply:
This also needs to be addressed.

Reply:
@selwin certainly - I have not had a chance to come back to this for some months.

The reason this was moved was tied to a change you commented on above - addressing https://github.com/rq/rq/pull/1170#discussion_r363012688 should make this change unnecessary.

Reply:
Moved back in c416694125755f58fbe2b771ee42894f6f201962 - you are correct; I've done my best to explain the approach taken in the commit message

> When a job with dependents is _successful_ it's dependents are enqueued. Only if the FINISHing job's `result_ttl` is non-zero is the change in status persisted in Redis - that is, when each dependent job is enqueued, the _FINISHing_ job (,triggering the enqueueing,) has an _outdated_ status in redis. This avoids redundant call because if `result_ttl=0` then the job is deleted then deleted in `Job#cleanup`.

> In order to enqueue the dependents, we therefore _exclude_ the FINISHing job from the check if each dependents' dependencies have been met.